Bug(`github-reposcaffolding-golang`): panic when execute `dtm delete -f xxx.yaml`

## Describe the Bug

Forget to set GITHUB_TOKEN env variable and execute the `./dtm-darwin-arm64 delete -y -f config-github-reposcaffolding-golang.yaml --debug`

```sh
...
...

2022-02-26 10:44:38 ℹ [INFO]  -------------------- [  Processing progress: 1/1.  ] --------------------
2022-02-26 10:44:38 ℹ [INFO]  Processing: crbot -> Delete ...
2022-02-26 10:44:38 ℹ [DEBUG]  Auth is enabled.
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0x0 pc=0x129758368]

goroutine 1 [running]:
github.com/merico-dev/stream/pkg/util/github.NewClient.func1()
        github.com/merico-dev/stream/pkg/util/github/github.go:45 +0x28
github.com/merico-dev/stream/pkg/util/github.NewClient(0x14000035a00)
        github.com/merico-dev/stream/pkg/util/github/github.go:74 +0x3fc
github.com/merico-dev/stream/internal/pkg/plugin/reposcaffolding/github/golang.uninstall(0x14000035980)
        github.com/merico-dev/stream/internal/pkg/plugin/reposcaffolding/github/golang/delete.go:36 +0x124
github.com/merico-dev/stream/internal/pkg/plugin/reposcaffolding/github/golang.Delete(0x14000035970)
        github.com/merico-dev/stream/internal/pkg/plugin/reposcaffolding/github/golang/delete.go:26 +0x194
github.com/merico-dev/stream/cmd/reposcaffolding/github/golang.Plugin.Delete({0x0, 0x0}, 0x14000035970)
        github.com/merico-dev/stream/cmd/reposcaffolding/github/golang/main.go:32 +0x34
github.com/merico-dev/stream/internal/pkg/pluginengine.Delete(0x14000035940)
        github.com/merico-dev/stream/internal/pkg/pluginengine/plugin.go:55 +0xf8
github.com/merico-dev/stream/internal/pkg/pluginengine.execute({0x100f88e10, 0x1400000e348}, {0x14000010228, 0x1, 0x1})
        github.com/merico-dev/stream/internal/pkg/pluginengine/change.go:104 +0x578
github.com/merico-dev/stream/internal/pkg/pluginengine.Remove({0x16f56f8f8, 0x29}, 0x1)
        github.com/merico-dev/stream/internal/pkg/pluginengine/cmd_delete.go:48 +0x39c
main.deleteCMDFunc(0x101286a20, {0x14000035400, 0x0, 0x4})
        github.com/merico-dev/stream/cmd/devstream/delete.go:23 +0x98
github.com/spf13/cobra.(*Command).execute(0x101286a20, {0x14000035380, 0x4, 0x4})
        github.com/spf13/cobra@v1.2.1/command.go:860 +0xa04
github.com/spf13/cobra.(*Command).ExecuteC(0x101286ca0)
        github.com/spf13/cobra@v1.2.1/command.go:974 +0x7b8
github.com/spf13/cobra.(*Command).Execute(0x101286ca0)
        github.com/spf13/cobra@v1.2.1/command.go:902 +0x30
main.main()
        github.com/merico-dev/stream/cmd/devstream/main.go:85 +0x30

```

--- a/internal/pkg/plugin/reposcaffolding/github/golang/golang.go
+++ b/internal/pkg/plugin/reposcaffolding/github/golang/golang.go
@@ -19,7 +19,7 @@ const (
 	DefaultTemplateRepo  = "dtm-scaffolding-golang"
 	DefaultTemplateOwner = "merico-dev"
 	TransitBranch        = "init-with-devstream"
-	MainBranch           = "main"
+	DefaultMainBranch    = "main"
 )
 
 type Config struct {
@@ -44,11 +44,14 @@ func InitRepoLocalAndPushToRemote(repoPath string, param *Param, ghClient *githu
 		return err
 	}
 
-	return MergeCommits(ghClient)
+	mainBranch := getMainBranchName(param)
+	return MergeCommits(ghClient, mainBranch)
 }
 
 func WalkLocalRepoPath(repoPath string, param *Param, ghClient *github.Client) error {
 	appName := param.Repo
+	mainBranch := getMainBranchName(param)
+
 	if err := filepath.Walk(repoPath, func(path string, info fs.FileInfo, err error) error {
 		if err != nil {
 			log.Debugf("Walk error: %s.", err)
@@ -95,13 +98,13 @@ func WalkLocalRepoPath(repoPath string, param *Param, ghClient *github.Client) e
 		} else {
 			// the main branch needs a initial commit
 			if strings.Contains(newPathForGithub, "gitignore") {
-				err := ghClient.CreateFile(content, strings.TrimSuffix(newPathForGithub, ".tpl"), MainBranch)
+				err := ghClient.CreateFile(content, strings.TrimSuffix(newPathForGithub, ".tpl"), mainBranch)
 				if err != nil {
 					log.Debugf("Failed to add the .gitignore file.")
 					return err
 				}
 				log.Debugf("Added the .gitignore file.")
-				return ghClient.NewBranch(MainBranch, TransitBranch)
+				return ghClient.NewBranch(mainBranch, TransitBranch)
 			}
 			return ghClient.CreateFile(content, strings.TrimSuffix(newPathForGithub, ".tpl"), TransitBranch)
 		}
@@ -112,8 +115,8 @@ func WalkLocalRepoPath(repoPath string, param *Param, ghClient *github.Client) e
 	return nil
 }
 
-func MergeCommits(ghClient *github.Client) error {
-	number, err := ghClient.NewPullRequest(TransitBranch, MainBranch)
+func MergeCommits(ghClient *github.Client, mainBranch string) error {
+	number, err := ghClient.NewPullRequest(TransitBranch, mainBranch)
 	if err != nil {
 		return err
 	}
@@ -185,4 +188,11 @@ func buildState(param *Param) map[string]interface{} {
 	res["owner"] = param.Owner
 	res["repoName"] = param.Repo
 	return res
+}
+
+func getMainBranchName(param *Param) string {
+	if param.Branch == "" {
+		return DefaultMainBranch
+	}
+	return param.Branch
 }

--- a/pkg/util/github/github.go
+++ b/pkg/util/github/github.go
@@ -41,7 +41,11 @@ func NewClient(option *Option) (*Client, error) {
 		return client, nil
 	}
 
+	var retErr error
 	defer func() {
+		if retErr != nil {
+			return
+		}
 		if client.Option.WorkPath == "" {
 			client.Option.WorkPath = DefaultWorkPath
 		}
@@ -71,8 +75,9 @@ func NewClient(option *Option) (*Client, error) {
 		token = os.Getenv("github_token")
 	}
 	if token == "" {
-		return nil, fmt.Errorf("failed to initialize GitHub token. More info - " +
+		retErr = fmt.Errorf("failed to initialize GitHub token. More info - " +
 			"https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/creating-a-personal-access-token")
+		return nil, retErr
 	}
 	log.Debugf("Token: %s.", token)
 

